File: tests/DES/ex_sdes.py
import sys, os
sys.path.append(os.path.abspath('../../src/DES'))
from sdes import *

secretkey = int(input('Enter the Secret Key(0-1023): '))
msg = input('Enter the message to be protected: ')
enc = ''
for i in msg:
    enc += chr(encrypt(ord(i), secretkey))
print('Encrypted message:')
print(enc)
dec =''
for i in enc:
    dec += chr(decrypt(ord(i), secretkey))
print('Decrypted message:')
print(dec)

File: tests/DES/ex_sdes.cpp
#include <iostream>
#include <cmath>
#include "../../src/DES/sdes.h"
using namespace std;

//  Function to display the binary format of a number
void printBin(int number, int exsize) {
    for(int i=1; i<=exsize; i++) {
        cout << (char)(number & (int)pow(2,exsize-i)?'1':'0');
    }
    cout << endl;
}

int main() {
    unsigned short int secretkey = 0b1010000010;
    uint8_t msg = 0b01110010;
    cout << "Original message:" << endl;
    printBin(msg,8);
    cout << "Secret key:" << endl;
    printBin(secretkey,10);
    uint8_t enc = encrypt(&msg, &secretkey);
    cout << "Encrypted message:" << endl;
    printBin(enc,8);
    uint8_t dec = decrypt(&enc, &secretkey);
    cout << "Decrypted message:" << endl;
    printBin(dec,8);
    return 0;
}
